src/Keboola/S3Extractor/S3ExceptionConverter.php

```diff
@@ -0,0 +1,75 @@
+<?php
+
+namespace Keboola\S3Extractor;
+
+use Aws\S3\Exception\S3Exception;
+use GuzzleHttp\Exception\ClientException;
+
+class S3ExceptionConverter
+{
+    /**
+     * @param S3Exception $e
+     * @throws Exception
+     * @throws S3Exception
+     */
+    public static function resolve(S3Exception $e): void
+    {
+        self::handleInvalidCredentials($e);
+        self::handleSlowDown($e);
+        self::handleBaseUserErrors($e);
+
+        throw $e;
+    }
+
+    /**
+     * @param S3Exception $e
+     * @throws Exception
+     */
+    private static function handleInvalidCredentials(S3Exception $e): void
+    {
+        if ($e->getStatusCode() === 403) {
+            throw new Exception('Invalid credentials or permissions.', $e->getCode(), $e);
+        }
+    }
+
+    /**
+     * @param S3Exception $e
+     * @throws Exception
+     */
+    private static function handleSlowDown(S3Exception $e): void
+    {
+        if ($e->getStatusCode() === 503) {
+            throw new Exception('Error 503 Slow Down: The number of requests to the S3 bucket is very high, please check your bucket limit.', $e->getCode(), $e);
```

Comment:
nevim jestli se ten limit da nekde checknout

Reply:
Vypadá to že ne, zkrátil jsem to.